core/src/components/checkbox/checkbox.tsx

```diff
@@ -289,7 +289,9 @@ Example: <ion-checkbox>Label</ion-checkbox>
 
 For checkboxes that do not have a visible label, developers should use "aria-label" so screen readers can announce the purpose of the checkbox.
 
-For checkboxes that do not render the label immediately next to the checkbox, developers may continue to use "ion-label" but must manually associate the label with the checkbox by using "aria-labelledby".`,
+For checkboxes that do not render the label immediately next to the checkbox, developers may continue to use "ion-label" but must manually associate the label with the checkbox by using "aria-labelledby".
+
+Developers can use the "legacy" property to continue using the legacy form markup. This property will be removed in an upcoming major release of Ionic where this form control will use the modern form markup.`,
```

Comment:
If we are good with the wording here, I can follow-up with another PR to apply across all form controls for v7.

Reply:
Looks good 👍